lib/Cake/Test/Case/Model/Datasource/DboSourceTest.php

```diff
@@ -1650,6 +1650,42 @@ public function testConditionKeysToString() {
 		$this->assertEquals($expected, $result[0]);
 	}
 
+	/**
+	 * Test conditionKeysToString() with virtual field
+	 *
+	 * @return void
+	 */
+	public function testConditionKeysToStringVirtualFieldExpression() {
+		$Article = ClassRegistry::init('Article');
+		$Article->virtualFields = array(
+			'extra' => DboSource::expression('something virtual')
```

Comment:
This isn't a static method. You need to use an instance like `$Articles->getDataSource()->expression()`.

Reply:
Sorry for the mistake. I will fix this soon